packages/nimble-components/src/table/specs/table-keyboard-navigation-hld.md

```diff
@@ -0,0 +1,180 @@
+
+2. The `Tab` key moves focus between the focusable elements in a row (not the cells). When focus has reached the end (or beginning of the row), a continued `Tab` (or `Shift-Tab`) will move focus outside of the table.
+3. When a cell has multiple interactive elements, or editable content, the user needs a way to leave navigation mode to interact with the cell (`Enter` and `F2` are suggested), and a way to return to navigation mode (`Esc`, also `Enter` for editable content).  
+   (Note: The `treegrid` and `grid` guidance are identical here.)
+
+If the above guidelines are adhered to, technologies like screen readers should be able to handle navigating through the table in an expected fashion.
+
+There's also guidance about whether cells themselves, or the elements inside the cells, should be focused:
+
+> ARIA: "For assistive technology users, the quality of experience when navigating a grid heavily depends on both what a cell contains and on where keyboard focus is set. For example, if a cell contains a button and a grid navigation key places focus on the cell instead of the button, screen readers announce the button label but do not tell users a button is present.  
+> There are two optimal cell design and focus behavior combinations:
+>
+> -   A cell contains one widget whose operation does not require arrow keys and grid navigation keys set focus on that widget. Examples of such widgets include link, button, menubutton, toggle button, radio button (not radio group), switch, and checkbox.
+> -   A cell contains text or a single graphic and grid navigation keys set focus on the cell."
+
+### Specifics for nimble-table
+
+#### Focusing the table, `Tab`/`Shift-Tab` behavior
+
+`Tab` or `Shift-Tab`, from neighboring elements on the page (according to tab order), will focus the table:
+
+-   If the table header contains any interactive/focusable elements, the 1st will be focused. Examples: 'Select all rows' checkbox, 'Collapse all groups' button
+-   Otherwise, the 1st table row will be focused
+-   (If no header elements are focusable, and the table has no rows, the table cannot be focused)
+
+As noted above, `Tab` / `Shift-Tab` key moves focus between the focusable elements in a row or the header (not the cells). When focus has reached the end (or beginning) of the row/header, a continued `Tab` (or `Shift-Tab`) will move focus outside of the table.
+
+Optionally, if a particular table row has been focused, then tab focus leaves the table and later returns, we may return focus to that row (rather than it restarting at the first row/ header).  
+**Open question**: Do we want to adopt this behavior?
```

Comment:
A few related questions:
- Would we return to the row based on ID or based on index?
- What would happen if the row doesn't exist anymore? (I assume we could just decide to fallback restarting focus back at the first row/header)
- Should we scroll the focused row back into view if it isn't currently in view?

Reply:
HLD updated:
- Current plan is to return to the row based on index, not ID. I think that makes the most sense since keyboard navigation is basically navigating by row index also.
- If the given row index no longer exists after an update, specified that we'll restart focus at the 1st row (not header)
- The focused row will be scrolled to, if needed